backend/logic/printful/sync/stockUpdate.js

```diff
@@ -0,0 +1,66 @@
+const _get = require('lodash/get')
+const program = require('commander')
+const { decryptConfig } = require('../../../utils/encryptedConfig')
+
+const { get } = require('../api')
+const { Product } = require('../../../models')
+
+/**
+ * Updates discontinued variants and variants that are out of stock to a collection.
+ * And creates collection if it doesn't exist
+ *
+ * @param {Object} shop
+ * @param {Number|String} productId
+ * @param {Array<Object>} variants
+ */
+const stockUpdate = async (shop, productId, variants) => {
+  const shopConfig = decryptConfig(shop.config)
+  const shopInventory = _get(shopConfig, 'inventory')
+  if (shopInventory) {
+    const variantsStock = variants.reduce(
+      (stockData, variant) => ({
+        ...stockData,
+        [variant]: 0
+      }),
+      {}
+    )
+
+    const apiKey = shopConfig.printful
+    let existingVariants = 0
+
+    if (apiKey) {
+      const productData = await get(`/store/products/${productId}`, {
```

Comment:
Why do we need to fetch the product data from Printful? Please add in the code to comment to explain. This is not obvious.

Reply:
I need to get all the variant's stock data because printful only sends out of stock and discontinued items. I will add that in the comments

Reply:
ok got it now. I did not realize Printful Sync does not pull the inventory data.sounds good.